File: Algorithm/javascript/BAEKJOON/CLASS 2/1018.js
function solution(input) {
  const [M, N] = input[0].split(" ").map((element) => parseInt(element));
  let board = [];
  let count = [];

  for (let i = 1; i <= M; i++) {
    board.push(input[i]);
  }
  // 8 X 8 크기의 체스판 생성

  for (x = 0; x < M - 7; x++) {
    for (y = 0; y < N - 7; y++) {
      blackStartCount = 0; // black 시작
      whiteStartCount = 0; // white 시작
      for (i = x; i < x + 8; i++) {
        for (j = y; j < y + 8; j++) {
          if ((i + j) % 2 === 0) {
            if (board[i][j] != "B") {
              // black 시작
              blackStartCount += 1;
            }
            // else if (board[i][j] != "W")
            else {
              // white 시작
              whiteStartCount += 1;
            }
          } else {
            if (board[i][j] != "W") {
              // black 시작
              blackStartCount += 1;
            }
            // else if (board[i][j] === "B")
            else {
              // white 시작
              whiteStartCount += 1;
            }
          }
        }
      }
      count.push(blackStartCount);
      count.push(whiteStartCount);
    }
  }
  // console.log(count);
  console.log(Math.min(...count));
}

const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on("line", (line) => {
  if (!line) {
    rl.close();
  } else {
    input.push(line);
  }
}).on("close", () => {
  solution(input);
  process.exit();
});

File: Algorithm/python/BAEKJOON/CLASS 2/1018.py
# board에서 W가 나온다음 "B"가 나와야 한다.
M, N = map(int, input().split(" "))

board = [input() for _ in range(M)]
count = []

# 처음 주어지는 크기와 상관없이 8 X 8 로 잘라야한다.
# M과 N에 -7을함으로써 추후에 올 for 문에서 조절이 가능
for x in range(M-7):
    for y in range(N-7):
        black_start_count = 0 # black 시작
        white_start_count = 0 # white 시작
        for i in range(x, x+8):
            for j in range(y, y+8):
                # i+j가 홀수이면, 해당 (i,j) 상하좌우는 짝수
                # (i, j) = (0, 0)이 체스판의 가장 왼쪽 상단(시작점)
                if (i+j) % 2 == 0: # i+j가 짝수인 경우(시작점 포함)
                    if board[i][j] != "B": # black 시작
                        black_start_count += 1
                    else: # white 시작
                        white_start_count += 1
                else: # i+j가 홀수인 경우
                    if board[i][j] != "W": # black 시작
                        black_start_count += 1
                    else: # white 시작
                        white_start_count += 1

        count += [black_start_count]
        count += [white_start_count]
print(min(count))
